src/olympia/abuse/tests/test_cinder.py

```diff
@@ -1567,6 +1569,56 @@ def test_workflow_recreate_waffle_switch_off(self):
         assert ActivityLog.objects.count() == 0
 
 
+class TestCinderAddonHandledByLegal(TestCinderAddon):
+    CinderClass = CinderAddonHandledByLegal
+    # For rendering the payload to Cinder like CinderAddon:
+    # - 1 Fetch Addon authors
+    # - 2 Fetch Promoted Addon
+    expected_queries_for_report = 2
+    expected_queue_suffix = None
+
+    def test_queue(self):
+        extension = self._create_dummy_target()
+        assert self.CinderClass(extension).queue == 'legal-escalations'
+
+    def test_queue_appeal(self):
+        extension = self._create_dummy_target()
+        assert self.CinderClass(extension).queue_appeal == 'legal-escalations'
+
+    def test_queue_with_theme(self):
+        # Contrary to reports handled by Cinder moderators, for reports handled
+        # by legal the queue should remain the same regardless of the
+        # addon-type.
+        target = self._create_dummy_target(type=amo.ADDON_STATICTHEME)
+        assert self.CinderClass(target).queue_appeal == 'legal-escalations'
+
+    def test_workflow_recreate(self):
+        addon = self._create_dummy_target()
+        listed_version = addon.current_version
+        listed_version.file.update(is_signed=True)
+        unlisted_version = version_factory(
+            addon=addon, channel=amo.CHANNEL_UNLISTED, file_kw={'is_signed': True}
+        )
+        ActivityLog.objects.all().delete()
```

Comment:
Question: why do we have to delete all activity logs? That will never happen in real life so if it's just to make the test pass, then there could be an issue in the setup?

Reply:
It's mainly because addon_factory and friends end up generating a lot of activity logs. We can't do `assert ActivityLog.objects.count() == 0` otherwise - we could filter by the specific activity log we expect not to find, but that leaves open the edge case of another activity log being generated of a different type.

`ActivityLog.objects.all().delete()` is a pretty common pattern though - maybe we should just do it in `addon_factory`? 🤷

Reply:
We can filter for the activity that's created for NeedsHumanReview., sure  Like I said though, it leaves open the edge case of other activity logs being generated of a different type, especially for future refactoring. Given we don't check anywhere for the presence of an ActivtyLog of a given type and do something different (outside of obvious places like the reviewer tools, where we may show "no activity", and gc/backfill/migrations), we aren't covering any edge cases by having extra log noise; but _are_ potentially opening an edge case by only checking for a particular type.

Reply:
This is still going over my head. 

> the edge case of other activity logs being generated of a different type

Why are we expecting that there is "no activity" here. It makes sense that we don't expect activity logs relating to NHR, which would match the narrow query I suggested. But I'm hearing you say there is an edge case that we could be creating other activity logs and I don't understand why we would care about that?

If you could explain why it's bad to have activity logs other than the NHR log, then I could get on board with it, otherwise it doesn't seem like an edge case worth protecting and our test is better fitted with a narrow query for the stuff that is actually important.

Reply:
🤷 not a bridge I'll die on, https://github.com/mozilla/addons-server/pull/22886/commits/a1d484f13855cac8d0fa13b4eced2052a10a876c